langchain4j-elasticsearch/src/main/java/dev/langchain4j/store/embedding/elasticsearch/ElasticsearchEmbeddingStore.java

```diff
@@ -269,6 +272,35 @@ public EmbeddingSearchResult<TextSegment> search(EmbeddingSearchRequest embeddin
         }
     }
 
+    @Override
+    public void remove(String id) {
+        ensureNotBlank(id, "id");
+        removeByIds(singletonList(id));
+    }
+
+    @Override
+    public void removeAll(Collection<String> ids) {
+        ensureNotEmpty(ids, "ids");
+        removeByIds(ids);
+    }
+
+    @Override
+    public void removeAll(Filter filter) {
+        Query query;
+        if (filter == null) {
```

Comment:
This is dangerous, please do `ensureNotNull(filter, "filter");`
Filter must be provided, otherwise it is a bug.
If user wants to remove all entries, (s)he can call `removeAll()`

Reply:
Fixed it. I followed the idea from `ElasticsearchEmbeddingStore.buildScriptScoreQuery` method in this same class, should I also fix that or open an issue?

Reply:
It is ok when there is no `Filter` in `search()` method, as filter is optional parameter there. But in `removeAll(Filter)` it is an only parameter and is supposed to be mandatory. If user did not provide it, there is a high chance that this might be a bug on user's side and (s)he is about to wipe out the entire store.